src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java

```diff
@@ -144,6 +145,11 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
                             try {
                                 if (future != null ) {
                                     listener.getLogger().println("Waiting for the completion of " + HyperlinkNote.encodeTo('/'+ p.getUrl(), p.getFullDisplayName()));
+                                    while(!future.isDone()) {
+                                        listener.getLogger().print(".");
+                                        Thread.sleep(SLEEP_MILLISECONDS);
```

Comment:
I think you will need to catch InterruptedException here and re-throw it. If the parent thread is interrupted (job aborted) during a blocking operation, it needs to be propagated, otherwise the job may never terminate.

Reply:
Ok thanks, I will look into this.